merfi/tests/backends/test_rpm_sign.py

```diff
@@ -1,40 +1,70 @@
-        # Fake the return values for inline-signing
-        m_util.run_output.return_value = ('--PGP SIGNED MESSAGE--', None, None)
-        super(TestRpmSign, self).test_sign_two_files(m_util, repotree)
-        detached_calls = [
-            call(self.detached),
-            call(self.detached),
-        ]
-        clearsign_calls = [
-            call(self.clearsign),
-            call(self.clearsign),
-        ]
-        m_util.run.assert_has_calls(detached_calls)
-        m_util.run_output.assert_has_calls(clearsign_calls)
+
+class RpmSign(object):
+
+    def setup(self):
+        self.backend = rpm_sign.RpmSign([])
+        self.backend.detached = CallRecorder()
+        self.backend.clear_sign = CallRecorder()
+        # fake command-line args
+        self.argv = ['merfi', '--key', 'mykey']
+        self.backend.parser = Transport(
+            self.argv, options=self.backend.options
+        )
+        self.backend.parser.parse_args()
+
+        # args to merfi.backends.rpm_sign.util's run()
+        self.detached = [
+            'rpm-sign', '--key', 'mykey', '--detachsign',
+            'Release', '--output', 'Release.gpg']
+        # args to merfi.backends.rpm_sign.util's run_output()
+        self.clearsign = [
+            'rpm-sign', '--key', 'mykey',
+            '--clearsign', 'Release']
+
+    def sign(self, directory):
+        self.backend.path = directory
+        self.backend.sign()
+
+
+class TestRpmSign(RpmSign):
+
+    def test_sign_no_files(self, tmpdir):
+        self.sign(str(tmpdir))
+        assert self.backend.detached.calls == []
+        assert self.backend.clear_sign.calls == []
+
+    def test_sign_two_files_detached_and_clearsign(self, repotree):
+        self.sign(repotree)
+        assert len(self.backend.detached.calls) == 2
+        assert len(self.backend.clear_sign.calls) == 2
+
+    def test_sign_two_files_path(self, repotree):
+        self.sign(repotree)
+        assert self.backend.detached.calls[0][0][0] == self.detached
+        assert self.backend.detached.calls[1][0][0] == self.detached
+
+    def test_sign_two_files_command(self, repotree):
+        self.sign(repotree)
+        assert self.backend.detached.calls[0][0][0] == self.detached
+        assert self.backend.detached.calls[1][0][0] == self.detached
+
+
+class TestRpmClearSign(RpmSign):
+
+    def test_sign_two_files_command(self, repotree):
+        self.sign(repotree)
+        # first call, second argument (the command to run)
+        assert self.backend.clear_sign.calls[0][0][1] == self.clearsign
+        # second call, second argument (the command to run)
+        assert self.backend.clear_sign.calls[1][0][1] == self.clearsign
+
+    def test_sign_two_files_path(self, repotree):
+        self.sign(repotree)
+        first_path = self.backend.clear_sign.calls[0][0][0]
+        second_path = self.backend.clear_sign.calls[1][0][0]
+        assert first_path.endswith('/dists/precise/Release')
```

Comment:
I'm getting a failure in this test:

```
$ py.test merfi/tests/
============================= test session starts ==============================
platform linux2 -- Python 2.7.10 -- py-1.4.30 -- pytest-2.7.3
rootdir: /home/kdreyer/dev/merfi, inifile: 
collected 24 items 

merfi/tests/test_base.py ..
merfi/tests/test_filecollector.py ......
merfi/tests/test_iso.py ..
merfi/tests/test_util.py ......
merfi/tests/backends/test_gpg.py ..
merfi/tests/backends/test_rpm_sign.py .....F

=================================== FAILURES ===================================
__________________ TestRpmClearSign.test_sign_two_files_path ___________________

self = <merfi.tests.backends.test_rpm_sign.TestRpmClearSign object at 0x7f0b8e3bfe90>
repotree = '/tmp/pytest-kdreyer/pytest-34/test_sign_two_files_path1'

    def test_sign_two_files_path(self, repotree):
        self.sign(repotree)
        first_path = self.backend.clear_sign.calls[0][0][0]
        second_path = self.backend.clear_sign.calls[1][0][0]
>       assert first_path.endswith('/dists/precise/Release')
E       assert <built-in method endswith of str object at 0x7f0b8e3c9198>('/dists/precise/Release')
E        +  where <built-in method endswith of str object at 0x7f0b8e3c9198> = '/tmp/pytest-kdreyer/pytest-34/test_sign_two_files_path1/dists/trusty/Release'.endswith

/home/kdreyer/dev/merfi/merfi/tests/backends/test_rpm_sign.py:69: AssertionError
----------------------------- Captured stdout call -----------------------------
--> Starting path collection, looking for files to sign
--> 2 matching paths found
--> will sign with the following commands:
--> rpm-sign --key "mykey" --detachsign Release --output Release.gpg
--> rpm-sign --key "mykey" --clearsign Release --output InRelease
--> signing: /tmp/pytest-kdreyer/pytest-34/test_sign_two_files_path1/dists/trusty/Release
--> signing: /tmp/pytest-kdreyer/pytest-34/test_sign_two_files_path1/dists/precise/Release
===================== 1 failed, 23 passed in 0.10 seconds ======================
```

I don't know why my computer sorts "trusty" before "precise" there so that "trusty" is `first_path`.

It doesn't really matter what order the calls are in, right? Maybe we should change this to simply be `endswith('Release')` ? Using the filename alone would have the added side effect of not hard-coding `os.sep` as `/` :)